data race possibility for `err` variable and `err` is allocated in heap

### Describe the bug

https://github.com/cli/cli/blob/8723e3bb5270af646fe7535b21e4db337a5bd73e/internal/codespaces/portforwarder/port_forwarder.go#L254-L316

https://github.com/cli/cli/blob/8723e3bb5270af646fe7535b21e4db337a5bd73e/internal/codespaces/portforwarder/port_forwarder.go#L277-L283

Two problems:
1. assigning `err`  in `go func()` defined in the outer scope may cause data race, although now it's protected by `done` channel synchronization.
2. `err` is allocated in heap, with small contribution to GC overhead. But it can be allocated in stack.

Evidence:

run `go build -a -gcflags=-m=2 ./... &> build.log` for commit 8723e3bb5270af646fe7535b21e4db337a5bd73e, you will see

```
...
internal/codespaces/portforwarder/port_forwarder.go:256:14: moved to heap: err
internal/codespaces/portforwarder/port_forwarder.go:258:20: ... argument does not escape
internal/codespaces/portforwarder/port_forwarder.go:258:20: &errors.errorString{...} escapes to heap
internal/codespaces/portforwarder/port_forwarder.go:269:20: ... argument does not escape
```

`err` is allocated in heap.

### Affected version

It seems that commit 581b665 does not solve the race problem.

### Steps to reproduce the behavior

It's detected by static analysis through a linter based on CodeQL with escape analysis extension([doc](https://github.com/Lslightly/QLStat/blob/main/doc/adapters/escape_analysis.md)).

### Expected vs actual behavior

Expected: In `go func`, use `err :=` instead of `err = `

### Logs

It's detected by static analysis.



<!-- Note: Set `GH_DEBUG=true` for verbose logs or `GH_DEBUG=api` for verbose logs with HTTP traffic details. -->

--- a/internal/codespaces/portforwarder/port_forwarder.go
+++ b/internal/codespaces/portforwarder/port_forwarder.go
@@ -276,7 +276,7 @@ func (fwd *CodespacesPortForwarder) UpdatePortVisibility(ctx context.Context, re
 	done := make(chan error)
 	go func() {
 		// Connect to the tunnel
-		err = fwd.connection.Connect(ctx)
+		err := fwd.connection.Connect(ctx)
 		if err != nil {
 			done <- fmt.Errorf("connect failed: %v", err)
 			return

